test/e2e/external_oidc_test.go

```diff
@@ -0,0 +1,461 @@
+	err = test.WaitForNewKASRollout(tc.t, ctx, tc.operatorConfigClient.OperatorV1().KubeAPIServers(), kasOrigRev)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func (tc *testClient) updateProxyForIngressCert(ctx context.Context) (cleanups []func(), err error) {
+	tc.t.Log("will copy default-ingress-cert and patch proxy")
+
+	defaultIngressCert, err := tc.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Get(ctx, "default-ingress-cert", metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+
+	cmCopy := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultIngressCert.Name,
+			Namespace: "openshift-config",
+		},
+		Data: defaultIngressCert.Data,
+	}
+
+	_, err = tc.kubeClient.CoreV1().ConfigMaps("openshift-config").Create(ctx, cmCopy, metav1.CreateOptions{})
+	cleanups = append(cleanups, func() {
+		err := tc.kubeClient.CoreV1().ConfigMaps("openshift-config").Delete(ctx, cmCopy.Name, metav1.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			tc.t.Fatalf("cleanup failed for configmap '%s/%s': %v", cmCopy.Namespace, cmCopy.Name, err)
+		}
+	})
+	if err != nil {
+		return
+	}
+
+	proxy, err := tc.configClient.ConfigV1().Proxies().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+
+	origTrustedCAName := proxy.Spec.TrustedCA.Name
+	proxy.Spec.TrustedCA.Name = "default-ingress-cert"
+	proxy, err = tc.configClient.ConfigV1().Proxies().Update(ctx, proxy, metav1.UpdateOptions{})
+	cleanups = append(cleanups, func() {
+		proxy.Spec.TrustedCA.Name = origTrustedCAName
+		proxy, err = tc.configClient.ConfigV1().Proxies().Update(ctx, proxy, metav1.UpdateOptions{})
+		if err != nil {
+			tc.t.Fatalf("cleanup failed for proxy '%s': %v", proxy.Name, err)
+		}
+	})
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func (tc *testClient) syncServingCA(ctx context.Context) (cleanups []func(), err error) {
+	if len(oidcCABundleConfigMap) == 0 || len(oidcCABundleConfigMapNamespace) == 0 {
+		tc.t.Log("no oidc CA defined; will use system CA")
+		return nil, nil
+	}
+
+	tc.t.Log("will sync OIDC ca-bundle to KAS static pod resources")
+
+	signingKey, err := tc.kubeClient.CoreV1().Secrets(oidcCABundleConfigMapNamespace).Get(ctx, oidcCABundleConfigMap, metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+
+	oidcServingCA := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "oidc-serving-ca",
+			Namespace: kasNamespace,
+		},
+		Data: map[string]string{
+			"ca-bundle.crt": string(signingKey.Data["tls.crt"]),
+		},
+	}
+
+	_, err = tc.kubeClient.CoreV1().ConfigMaps(kasNamespace).Create(ctx, oidcServingCA, metav1.CreateOptions{})
+	cleanups = append(cleanups, func() {
+		err := tc.kubeClient.CoreV1().ConfigMaps(kasNamespace).Delete(ctx, oidcServingCA.Name, metav1.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			tc.t.Fatalf("cleanup failed for secret '%s/%s': %v", oidcServingCA.Namespace, oidcServingCA.Name, err)
+		}
+	})
+	if err != nil {
+		return
+	}
+
+	return
```

Comment:
Seems only creating a configmap is not enough. We need also make the "oidc-serving-ca" configmap mounted in the kube-apiserver pods, right? I'm not sure currently whether there is a way to achieve the mounting [1] : )
[1] I tried mounting as below, but the kube-apiserver-asahay02417-m2j57-master-0 never appears:
```
On one control plane node master-0, modify /etc/kubernetes/manifests/kube-apiserver-pod.yaml to append below volumeMounts and volumes:
{"spec":{"containers":[{"name":"kube-apiserver","volumeMounts":[...,{"mountPath":"/etc/kubernetes/static-pod-resources/configmaps/oidc-serving-ca","name":"oidc-ca"}]}],"volumes":[...,{"configMap":{"defaultMode":420,"name":"oidc-serving-ca","items":[{"key":"ca-bundle.crt","path":"ca-bundle.crt"}]},"name":"oidc-ca"}]}}

The kube-apiserver-asahay02417-m2j57-master-0 never appears:
$ oc get po -n openshift-kube-apiserver -l apiserver
NAME                                        READY   STATUS    RESTARTS   AGE
kube-apiserver-asahay02417-m2j57-master-1   5/5     Running   0          3h28m
kube-apiserver-asahay02417-m2j57-master-2   5/5     Running   0          3h34m
```

Reply:
You are right @xingxingxia -- we will need something like this: https://github.com/openshift/cluster-kube-apiserver-operator/pull/1720

For now this functionality isn't actually needed for the test, it's more of a placeholder for when we further develop this feature. I'll add a comment to the code with a link to that PR.

Reply:
Fixed in 0e62f4e60d8277f43443976f4fefcc98909f8af5